### PR TITLE
device identity

### PR DESCRIPTION
**Pull request by piotrbartman**
implements QubesOS/qubes-issues/issues/9325

### CODE REVIEW

**Comment by piotrbartman** [comment]
only error is: `No name 'device_protocol' in module 'qubesadmin' (no-name-in-module)` which could be ignored, since `qubesadmin` is changed in QubesOS/qubes-core-admin-client/pull/304

**Comment by piotrbartman** [comment]
pylint complains about changes not related to this PR

File: qubesmanager/settings.py [hunk 1]
```diff
@@ -1270,32 +1270,32 @@ def __apply_devices_tab__(self):
...
+                    if dev.port_id in self.new_strict_reset_list:
                         options['no-strict-reset'] = True
-                    ass = devices.DeviceAssignment(
-                        self.vm.app.domains['dom0'],
-                        dev.ident, devclass='pci',
-                        attach_automatically=True, required=True,
+                    ass = device_protocol.DeviceAssignment(
+                        device_protocol.Port(
+                            self.vm.app.domains['dom0'], dev.port_id, 'pci'),
+                        mode='required',
```
Comment (marmarek) [inline qubesmanager/settings.py:1278]: This does not match new DeviceAssignment constructor after https://github.com/QubesOS/qubes-core-admin-client/pull/304/commits/8772108b263a3379a3c53cf8b6cb14576889bb7d#diff-3c9f34f047553a6c2bc231786134cfece1c2d94ba3931a216f6a91840e130e95L860-R1015

Maybe you want to use the DeviceAssignment.new helper here?

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by marmarek** [comment]
See https://github.com/QubesOS/qubes-manager/pull/395/commits/338c5ed130fc2fc979e72d49466f5566374d5724, there is also an issue with listing assigned devices.

**Review by marmarek — APPROVED** [review]
This is seems to work good enough, together with #395, lets handle minor issues in follow up PRs

**Pull request merged**